ios/RNSScreenStack.mm

```diff
@@ -1176,16 +1176,7 @@ - (nullable RNSScreenView *)childScreenForTag:(react::Tag)tag
 - (void)unmountChildComponentView:(UIView<RCTComponentViewProtocol> *)childComponentView index:(NSInteger)index
 {
   RNSScreenView *screenChildComponent = (RNSScreenView *)childComponentView;
-
-  // We should only do a snapshot of a screen that is on the top.
-  // We also check `_presentedModals` since if you push 2 modals, second one is not a "child" of _controller.
-  // Also, when dissmised with a gesture, the screen already is not under the window, so we don't need to apply
-  // snapshot.
-  if (screenChildComponent.window != nil &&
-      ((screenChildComponent == _controller.visibleViewController.view && _presentedModals.count < 2) ||
-       screenChildComponent == [_presentedModals.lastObject view])) {
-    [screenChildComponent.controller setViewToSnapshot];
-  }
+  [screenChildComponent.controller setViewToSnapshot];
```

Comment:
I see that this simplifies the logic, but I don't think this is quite right. 

Imagine you have Stack with three screens: "A, B, C". You pop from C to top, thus resulting stack is "A". This code would do snapshots of both C and B, while we only want to do a snapshot of C.

If you describe what problem exactly you intend to solve we might be able to figure out better solution.

Reply:
Will this code: https://github.com/software-mansion/react-native-screens/blob/46147026c708eab19de8082a2d2a1c7fd680b30c/ios/RNSScreen.mm#L1842 return `true` for `B`? I think its view is not in the window since it is detached. Would have to check though.

Reply:
I checked. It is `nil` then so it should work correctly.

Reply:
+1, I've forgotten that UINavigationController will keep all the controllers lower in the stack in detached state and that's it why it will work correctly.